### PR TITLE
add Probot auto-merge [wip]

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

experimental config for adding auto merging, pls see description - https://github.com/bobvanderlinden/probot-auto-merge

### Alternatives:
- https://github.com/palantir/bulldozer
- https://github.com/marketplace/actions/merge-pull-requests
- https://github.com/mergeability/mergeable

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+minApprovals:
+  COLLABORATOR: 3
+# The maximum number of reviews from each association that request changes to the pull request.
+# Setting this number higher than 0 will allow automatic merging while changes are still requested.
+# Requested changes from associations not defined in this list are ignored for automatic merging.
+maxRequestedChanges:
+  NONE: 0
+
+# Automatic merges will be blocked if there is a match between the regular expression and title
+blockingTitleRegex: '\bwip\b'
```
Comment (justusschock) [inline .github/auto-merge.yml:30]: can we also have a blocked label or add this to the regex here? This is useful when having multiple dependant PRs

File: .github/auto-merge.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# Note: this only works when the branch of the pull request resides in the same repository as
+#       the pull request itself.
+deleteBranchAfterMerge: true
+reportStatus: true
+
+# The minimum number of reviews from each association that approve the pull request before
+# doing an automatic merge. For more information about associations see:
+# https://developer.github.com/v4/enum/commentauthorassociation/
+minApprovals:
+  COLLABORATOR: 3
```
Comment (justusschock) [inline .github/auto-merge.yml:22]: shouldn't we also add owner approval here?

**Comment by ydcjeff** [comment]
wait a bit and shall we use github native PR auto-merge?
https://github.blog/2020-12-08-new-from-universe-2020-dark-mode-github-sponsors-for-companies-and-more/#pull-request-auto-merge

**Comment by Borda** [comment]
seems that the `requiredChecks` is missing https://github.com/bobvanderlinden/probot-auto-merge/issues/312#issuecomment-745311656

File: .github/auto-merge.yml (hunk 2 above)
Comment (Borda) [reply .github/auto-merge.yml:22]: it shall in default, but true that it is no very clear hos the rules in root and here works together...

**Comment by Borda** [comment]
> wait a bit and shall we use github native PR auto-merge?
> https://github.blog/2020-12-08-new-from-universe-2020-dark-mode-github-sponsors-for-companies-and-more/#pull-request-auto-merge

can you pls check for some more details about the configuration and if we van set it in an extra file (outside GitHub UI)

**Comment by ydcjeff** [comment]
> > wait a bit and shall we use github native PR auto-merge?
> > https://github.blog/2020-12-08-new-from-universe-2020-dark-mode-github-sponsors-for-companies-and-more/#pull-request-auto-merge
> 
> can you pls check for some more details about the configuration and if we van set it in an extra file (outside GitHub UI)

as far as I found out, it can only be set via github UI

**Comment by Borda** [comment]
let's try native GH auto-merge and eventually resume this one...

**Pull request closed without merging**